cpp/src/io/functions.cpp

```diff
@@ -338,6 +338,34 @@ parsed_orc_statistics read_parsed_orc_statistics(source_info const& src_info)
   return result;
 }
 
+orc_column_metadata make_orc_column_metadata(host_span<orc::SchemaType const> orc_schema,
+                                             uint32_t column_id,
+                                             std::string column_name)
+{
+  orc_column_metadata col_meta{column_name, orc_schema[column_id].kind};
+  auto const& orc_col_schema = orc_schema[column_id];
+  for (auto i = 0ul; i < orc_col_schema.subtypes.size(); ++i) {
+    col_meta.children.push_back(make_orc_column_metadata(
+      orc_schema,
+      orc_col_schema.subtypes[i],
+      i < orc_col_schema.fieldNames.size() ? orc_col_schema.fieldNames[i] : ""));
+  }
+
+  return col_meta;
+}
+
+orc_metadata read_orc_metadata(source_info const& src_info)
```

Comment:
Same const question here.

Reply:
also encapsulated, as silly as it looks ATM.